### PR TITLE
Improve error message on systemd-run failure

### PR DESCRIPTION
**Pull request by pgombar**
## Description

Make it clear that the extension operation is immediately retried without using systemd in case systemd-run fails.

Old log entry:
```
2019/10/30 02:02:58.967870 ERROR ExtHandler Event: name=WALinuxAgent, op=InvokeCommandUsingSystemd, message=Failed to run systemd-run for unit Microsoft.CPlat.Core.RunCommandLinux_1.0.1_ef8202d8-8322-4c3f-bdfe-162c49b6d98a.scope. Systemd process exited with code 1 and output [stdout]


[stderr]
Failed to start transient scope unit: Connection timed out
, duration=0
```

New log entry:
```
2019/10/30 02:02:58.967870 ERROR ExtHandler Failed to run systemd-run for unit Microsoft.CPlat.Core.RunCommandLinux_1.0.1_ef8202d8-8322-4c3f-bdfe-162c49b6d98a.scope. Retrying extension operation without systemd. Error: Systemd process exited with code 1 and output [stdout]


[stderr]
Failed to start transient scope unit: Connection timed out
```

---

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes in the description.
- [ ] New Unit tests were added for the changes made and Travis.CI is passing.

### Quality of Code and Contribution Guidelines
- [ ] I have read the [contribution guidelines](https://github.com/Azure/WALinuxAgent/blob/master/.github/CONTRIBUTING.md).

---
This change is [Reviewable](https://reviewable.io/reviews/azure/walinuxagent/1699)

### CODE REVIEW

File: azurelinuxagent/common/cgroupapi.py [hunk 1]
```diff
@@ -514,12 +514,16 @@ def create_cgroup(controller):
...
+                            'Retrying extension operation without systemd. Error: {1}'.format(scope_name, err_msg)
                 add_event(AGENT_NAME,
                           version=CURRENT_VERSION,
                           op=WALAEventOperation.InvokeCommandUsingSystemd,
                           is_success=False,
-                          message='Failed to run systemd-run for unit {0}.scope. {1}'
-                                  .format(scope_name, err_msg))
+                          log_event=False,
+                          message=event_msg)
+                logger.error(event_msg)
```
Comment (narrieta) [inline azurelinuxagent/common/cgroupapi.py]: warning?

File: azurelinuxagent/common/cgroupapi.py [hunk 2]
```diff
@@ -514,12 +514,16 @@ def create_cgroup(controller):
                 # There was an issue with systemd-run. We need to log it and retry the extension without systemd.
                 err_msg = 'Systemd process exited with code %s and output %s' % (e.exit_code, process_output) \
                     if isinstance(e, ExtensionOperationError) else "Systemd timed-out, output: %s" % process_output
+                event_msg = 'Failed to run systemd-run for unit {0}.scope. ' \
```
Comment (narrieta) [inline azurelinuxagent/common/cgroupapi.py]: Somehow, the way I read your sample error message, it gives me the impression that the error is coming from the retry.

Maybe 

> Failed to run systemd-run for unit {0}.scope. Will retry invoking the extension without systemd. Systemd error: {1}

File: azurelinuxagent/common/cgroupapi.py (hunk 1 above)
Comment (pgombar) [reply azurelinuxagent/common/cgroupapi.py]: Agree, error is too strong considering we try to recover.

File: azurelinuxagent/common/cgroupapi.py (hunk 2 above)
Comment (pgombar) [reply azurelinuxagent/common/cgroupapi.py]: That sounds better, updating.

**Pull request merged**